src/installer.js

```diff
@@ -10,28 +13,66 @@ const path = require('path')
  * @param {string[]} hexMirrors
  */
 async function installOTP(osVersion, otpVersion, hexMirrors) {
+  if (hexMirrors.length === 0) {
+    throw new Error(
+      `Could not install Erlang/OTP ${otpVersion} from any hex.pm mirror`,
+    )
+  }
+
+  const [hexMirror, ...hexMirrorsT] = hexMirrors
+  const fullVersion = `${osVersion}/${otpVersion}`
+  let cachedPath = tc.find('otp', fullVersion)
   const OS = process.platform
-  if (OS === 'linux') {
-    if (hexMirrors.length === 0) {
-      throw new Error(
-        `Could not install Erlang/OTP ${otpVersion} from any hex.pm mirror`,
-      )
-    }
-    const [hexMirror, ...hexMirrorsT] = hexMirrors
-    try {
-      await exec(path.join(__dirname, 'install-otp.sh'), [
-        osVersion,
-        otpVersion,
-        hexMirror,
-      ])
-      return
-    } catch (err) {
-      core.info(`install-otp.sh failed for mirror ${hexMirror}`)
+
+  try {
+    if (OS === 'linux') {
+      if (!cachedPath) {
+        const tarPath = await tc.downloadTool(
+          `https://builds.hex.pm/builds/otp/${fullVersion}.tar.gz`,
+        )
+        const extractPath = await tc.extractTar(tarPath, undefined, [
+          'zx',
+          '--strip-components=1',
+        ])
+        cachedPath = await tc.cacheDir(extractPath, 'otp', fullVersion)
+      }
+
+      // TODO: Can we cache install?
```

Comment:
If this is not to be solved in the context of this pull request, could you create a task to track it, and delete the comment? I think there's a similar one below. What's the measurable improvement in caching the installation too?

Reply:
There's no improvement unless you do multiple installations in one workflow run. I don't think anyone is doing that.

Reply:
Perfect!